recipes/openexr/all/conanfile.py

```diff
@@ -0,0 +1,79 @@
+from conans import ConanFile, CMake, tools
+import os
+class OpenEXRConan(ConanFile):
+    name = "openexr"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/openexr/openexr"
+    topics = ("conan")
+    license = "BSD-3"
+    description = "OpenEXR is a high dynamic-range (HDR) image file format for use in computer imaging applications"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False], "namespace_versioning": [True, False]}
+    default_options = {"shared": False, "namespace_versioning": True, 'fPIC': True}
+    generators = "cmake", "cmake_find_package"
+    exports_sources = ["CMakeLists.txt", "patches/0001-find-zlib.patch"]
+    _source_subfolder = "source_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_folder = "openexr-{}".format(self.version)
+        os.rename(extracted_folder, self._source_subfolder)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            self.options.remove("fPIC")
+
+    def requirements(self):
+        self.requires.add("zlib/1.2.11")
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["OPENEXR_BUILD_PYTHON_LIBS"] = False
+        cmake.definitions["BUILD_TESTING"] = False
+        cmake.definitions["OPENEXR_VIEWERS_ENABLE"] = False
+        cmake.definitions["OPENEXR_BUILD_SHARED"] = self.options.shared
+        cmake.definitions["OPENEXR_BUILD_STATIC"] = not self.options.shared
+        cmake.definitions["OPENEXR_NAMESPACE_VERSIONING"] = self.options.namespace_versioning
+        cmake.definitions["OPENEXR_ENABLE_TESTS"] = False
+        cmake.definitions["OPENEXR_FORCE_CXX03"] = True
+        cmake.definitions["OPENEXR_BUILD_UTILS"] = False
+        cmake.definitions["ENABLE_TESTS"] = False
+        cmake.definitions["OPENEXR_BUILD_TESTS"] = False
+
+        cmake.configure(source_folder=self._source_subfolder)
+        return cmake
+
+    def build(self):
+        with tools.chdir(self.source_folder):
+            tools.patch(base_path=self._source_subfolder, patch_file="patches/0001-find-zlib.patch")
```

Comment:
As your are using the generic version folder `all`, consider using conandata to apply it:

https://github.com/conan-io/hooks/pull/109#issue-328338976